lib/invalidScopeDisables.js

```diff
@@ -6,10 +6,10 @@
 
 /**
  * @param {import('stylelint').StylelintResult[]} results
- * @param {import('stylelint').StylelintConfig} config
+ * @param {import('stylelint').StylelintConfig|undefined} config
```

Comment:
```suggestion
 * @param {import('stylelint').StylelintConfig} config
```

It works without `undefined`.

Reply:
In this case type annotation in JSDoc is not correct, it means the function accepts `config` only with `StylelintConfig` type, but it's not true. I think it's more correct to specify in type that function may accept `undefined`. Also, we can describe `config` as optional - https://jsdoc.app/tags-param.html#optional-parameters-and-default-values, but I think it's a worse approach.

Reply:
> By default `null` and `undefined` are subtypes of all other types. That means you can assign `null` and `undefined` to something like `number`.

http://www.typescriptlang.org/docs/handbook/basic-types.html#null-and-undefined

While explicitly specifying `undefined` is correct, it's not necessary.

If you want to keep, it's ok.